lib/widget/widgbase.h

```diff
@@ -191,12 +191,16 @@ class WIDGET
 
 	WIDGET *parent()
 	{
-		return parentWidget;
+		return parentWidget.lock().get();
```

Comment:
Unfortunately, the call to `lock()` creates a temporary `shared_ptr` which no longer owns a strong reference to the underlying WIDGET after the call to `parent()` returns (as it is destroyed), so we probably need to change `parent()` to return a `std::shared_ptr<WIDGET>` for safety. (Or, theoretically, it can return a `weak_ptr`, but that just passes the buck for locking to callers - so I'd argue for `shared_ptr<WIDGET>` as the return value.)

Reply:
Changed `parent()` to return `shared_ptr`.